cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -1436,6 +1440,32 @@ def make_resource(self, data, **kwargs):
         """Generate resource."""
         return SchedulerPluginSettings(**data)
 
+    @pre_load
+    def fetch_scheduler_definition(self, data, **kwargs):
+        """Fetch scheduler definition if it is s3 or https url."""
+        original_scheduler_definition = data["SchedulerDefinition"]
+        if type(original_scheduler_definition) is str:
+            try:
+                if original_scheduler_definition.startswith("s3"):
+                    bucket_parsing_result = parse_bucket_url(original_scheduler_definition)
```

Comment:
what if the string starts with s3 but then the format is wrong, is the error message clear enough? same for https.

Reply:
if it is not valid format of s3,  it'll raise an error (Invalid s3 url: {url}) and be caught in Exception, throw ValidationError with wrapped error message. For https, if it is invalid url, it'll raise urlopen error and be caught in Exception as well.